services/user.py

```diff
@@ -0,0 +1,64 @@
+from typing import Optional
+
+from django.contrib.auth import get_user_model
+
+from db.models import User
+
+
+def create_user(
+        username: str,
+        password: str,
+        email: Optional[str] = None,
+        first_name: Optional[str] = None,
+        last_name: Optional[str] = None
+) -> User:
+
+    created_user = get_user_model().objects.create_user(
+        username=username,
+        password=password,
+    )
+    updated_user = get_user_model().objects.get(username=username)
+    if email:
+        updated_user.email = email
+
+    if first_name:
+        updated_user.first_name = first_name
+
+    if last_name:
+        updated_user.last_name = last_name
+
+    updated_user.save()
```

Comment:
Don't update user after creation. Just put all the needed values into `create_user` method.